File: Sistema para Bibliotecas Públicas/Gerenciador Biblioteca/Biblioteca/Models/Emprestimo.cs
namespace Biblioteca.Models
{
    public class Emprestimo
    {
        public Emprestimo()
        {
            EmprestimoId = Guid.NewGuid().ToString();
        }

        public string? EmprestimoId { get; set; }

        public string? LivroId { get; set; }

        public Livro Livro { get; set; }

        public string LeitorId { get; set; }

        public Leitor Leitor { get; set; }

        public DateTime DataEmprestimo { get; set; } = DateTime.Now;
        
        public DateTime PrazoDevolucao { get; set; } = DateTime.Now.AddDays(14);

        public DateTime DataDevolucao  { get; set; }

        public bool Ativo { get; set; } = true;
    }
}

File: Sistema para Bibliotecas Públicas/my-app/src/models/Emprestimo.ts
import { Autor } from "./Autor";
import { Leitor } from "./Leitor";
import { Livro } from "./Livro";

export interface Emprestimo {
    emprestimoId: string;
    dataEmprestimo: string;
    prazoDevolucao: string;
    dataDevolucao: string;
    ativo: boolean;
    livroId: string;
    autorId: string;
    leitorId: string;
    autor: Autor;
    leitor: Leitor;
    livro: Livro;
}
